### PR TITLE
feat: support jina ai embedding and reranker

### PR DESCRIPTION
**Pull request by ZiniuYu**
Support for Jina AI Embedding and Reranker via API:

- Integrated Jina AI's embedding and reranking features through the API.
  - Ensured compatibility and optimized performance for real-time data processing.
- Added documentation to guide users on how to utilize the new Jina AI embedding and reranking features.

### CODE REVIEW

File: apps/docs/docs/modules/node_postprocessors/jinaai_reranker.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+retriever.similarityTopK = 5;
+```
+
+## Create a new instance of the JinaAIReranker class
+
+Then you can create a new instance of the `JinaAIReranker` class and pass in the number of results you want to return.
+The Jina AI Reranker API key is set in the `JINAAI_API_KEY` environment variable.
+
+```bash
+export JINA_API_KEY=<YOUR API KEY>
```
Comment (DresAaron) [inline apps/docs/docs/modules/node_postprocessors/jinaai_reranker.md:52]: `export JINAAI_API_KEY` or `JINA_API_KEY`?

**Review by DresAaron — CHANGES_REQUESTED** [review]

File: packages/core/src/postprocessors/rerankers/JinaAIReranker.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  relevance_score: number;
+}
+
+export class JinaAIReranker implements BaseNodePostprocessor {
+  model: string = "jina-reranker-v1-base-en";
+  topN?: number;
+  apiKey?: string = undefined;
+
+  constructor(init?: Partial<JinaAIReranker>) {
+    this.topN = init?.topN;
```
Comment (Mini256) [inline packages/core/src/postprocessors/rerankers/JinaAIReranker.ts:20]:
```suggestion
    this.topN = init?.topN ?? 2;
```

Should we add default value for the topN parameter, like python version?

https://github.com/run-llama/llama_index/blob/aa13d47444692faa06b5753b7451b1920837b29c/llama-index-integrations/postprocessor/llama-index-postprocessor-jinaai-rerank/llama_index/postprocessor/jinaai_rerank/base.py#L26

Comment (ZiniuYu) [reply packages/core/src/postprocessors/rerankers/JinaAIReranker.ts:20]: Thanks for the comment! In fact we do not require topN in the api, the default value will be len(docs) if it's not provided. I will discuss whether we want a good value for it with my colleagues.

**Review by marcusschiesser — APPROVED** [review]

**Comment by marcusschiesser** [comment]
@ZiniuYu Looks great, thanks. Can you call `pnpx changeset` and add a changeset? Then your change will end up in our changelog.md file

**Comment by ZiniuYu** [comment]
@marcusschiesser Thanks for reviewing! The changeset is added

**Pull request merged**